src/shared/dns-domain.c

```diff
@@ -792,9 +790,8 @@ int dns_name_address(const char *p, int *ret_family, union in_addr_union *ret_ad
                 return r;
         if (r > 0) {
                 struct in6_addr a;
-                unsigned i;
 
-                for (i = 0; i < ELEMENTSOF(a.s6_addr); i++) {
+                for (unsigned i = 0; i < ELEMENTSOF(a.s6_addr); i++) {
```

Comment:
might as well switch to size_t, after all that's what ELEMENTSOF() returns. pointless losing 32bit here for no reason. (same above)

Reply:
Yes, but we know that the maximum number we use is around 8… I'll change it, but I think it's pointless in this case.